server/datastore/mysql/apple_mdm_test.go

```diff
@@ -4396,6 +4397,60 @@ func testMDMAppleDeleteHostDEPAssignments(t *testing.T, ds *Datastore) {
 	}
 }
 
+func testCleanMacOSMDMLock(t *testing.T, ds *Datastore) {
+	ctx := context.Background()
+
+	checkState := func(t *testing.T, status *fleet.HostLockWipeStatus, unlocked, locked, wiped, pendingUnlock, pendingLock, pendingWipe bool) {
```

Comment:
Nit: I think I moved a similar function to the file level (not an inner function anymore), you could reuse that. We can address that post-freeze.